File: GetURL/AppiumTest.java
import java.io.*;
import java.net.*;
import java.util.*;
import java.util.concurrent.TimeUnit;

import org.openqa.selenium.*;
import org.openqa.selenium.html5.*;
import org.openqa.selenium.logging.*;
import org.openqa.selenium.remote.*;
import org.openqa.selenium.Cookie.Builder;

import io.appium.java_client.*;
import io.appium.java_client.android.*;
import io.appium.java_client.ios.*;

public class AppiumTest {

    public static void main(String[] args) throws MalformedURLException, IOException {
        System.out.println("Run started");

        String browserName = "mobileOS";
        DesiredCapabilities capabilities = new DesiredCapabilities("", "", Platform.ANY);
        String host = "your cloud";
        capabilities.setCapability("user", "your cloud username");
        capabilities.setCapability("password", "your cloud password");

        //TODO: Change your device ID
        capabilities.setCapability("deviceName", "DeviceID");

        // Use the automationName capability to define the required framework - Appium (this is the default) or PerfectoMobile.
        capabilities.setCapability("automationName", "Appium");

        // Call this method if you want the script to share the devices with the Perfecto Lab plugin.
        PerfectoLabUtils.setExecutionIdCapability(capabilities, host);

        // Application settings examples.
        capabilities.setCapability("app", "PUBLIC:appium/MultipleWebViews.ipa");
		capabilities.setCapability("autoInstrument", true);
        // For Android:
        // capabilities.setCapability("appPackage", "com.google.android.keep");
        // capabilities.setCapability("appActivity", ".activities.BrowseActivity");
        // For iOS:
        capabilities.setCapability("bundleId", "42S4GZUXGS.MultipleWebViews");

        // Add a persona to your script (see https://community.perfectomobile.com/posts/1048047-available-personas)
        //capabilities.setCapability(WindTunnelUtils.WIND_TUNNEL_PERSONA_CAPABILITY, WindTunnelUtils.GEORGIA);

        // Name your script
        capabilities.setCapability("scriptName", "Testing getCurrentUrl");

        //AndroidDriver driver = new AndroidDriver(new URL("https://" + host + "/nexperience/perfectomobile/wd/hub"), capabilities);
        IOSDriver driver = new IOSDriver(new URL("https://" + host + "/nexperience/perfectomobile/wd/hub"), capabilities);
        driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);

        By topWindow = By.xpath("//*[@label='address1']");
        By btmWindow = By.xpath("//*[@label='address2']");
        By goBtn1 = By.xpath("//*[@label='button_navigate1']");
        By goBtn2 = By.xpath("//*[@label='button_navigate2']");
        
        try {
            // write your code here
        	// send top window to espn
        	driver.context("NATIVE_APP");
        	driver.findElement(topWindow).sendKeys("www.espn.com");
        	driver.findElement(goBtn1).click();
        	
        	// send bottom window to cbsSports
        	driver.findElement(btmWindow).sendKeys("www.cbssports.com");
        	driver.findElement(goBtn2).click();
        	

        	// set driver context to top webview and retrieve URL
        	driver.context("WEBVIEW_1");
        	System.out.println(driver.getCurrentUrl());
        	
        	driver.context("WEBVIEW_2");
        	System.out.println(driver.getCurrentUrl()); 
        	
        	driver.context("WEBVIEW");
        	System.out.println(driver.getCurrentUrl()); 

        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            try {
                // Retrieve the URL of the Single Test Report, can be saved to your execution summary and used to download the report at a later point
                String reportURL = (String)(driver.getCapabilities().getCapability(WindTunnelUtils.SINGLE_TEST_REPORT_URL_CAPABILITY));

                driver.close();

                // In case you want to download the report or the report attachments, do it here.
                PerfectoLabUtils.downloadReport(driver, "pdf", "C:\\test\\reportGetUrl");
                // PerfectoLabUtils.downloadAttachment(driver, "video", "C:\\test\\report\\video", "flv");
                // PerfectoLabUtils.downloadAttachment(driver, "image", "C:\\test\\report\\images", "jpg");

            } catch (Exception e) {
                e.printStackTrace();
            }

            driver.quit();
        }

        System.out.println("Run ended");
    }
}

File: Appium/python/AppiumTest.py
from appium import webdriver
from RemoteWebDriverUtils import RemoteWebDriverUtils

class AppiumTest():
            
    def main(self):
        
        # Create a desired capabilities object as a starting point.
        capabilities = {}
        capabilities['version'] = ''
        capabilities['browserName'] = ''
        capabilities['platform'] = 'ANY'
        host = 'myHost.perfectomobile.com'
        capabilities['user'] = 'myUser'
        capabilities['password'] = 'myPassword'
        capabilities['automationName'] = 'Appium'
        
        # Call this method if you want the script to share the devices with the recording plugin.
        #RemoteWebDriverUtils.set_execution_id_capability(capabilities)
        
        # Application settings examples.
        #     capabilities["app"] = "PRIVATE:applications/Errands.ipa"
        # For Android:
        #     capabilities['platformName'] = 'Android'
        #     capabilities['platformVersion'] = '5.*' 
        #     capabilities["appPackage"] = "com.google.android.keep"
        #     capabilities["appActivity"] = ".activities.BrowseActivity"
        # For iOS:
        #     capabilities['platformName'] = 'iOS'
        #     capabilities['platformVersion'] = '8|9.*' 
        #     capabilities["bundleId"] = "com.yoctoville.errands"
        
        driver = webdriver.Remote("https://" + host + "/nexperience/perfectomobile/wd/hub", capabilities)
        
        try :
            # write your code here...
            print('Driver created')
            
        except Exception as e:
            print(e)
        finally:
            # disconnect from the Remote server
            if driver is not None:
                driver.close()
                
                # In case you want to down the report or the report attachments, do it here.
                RemoteWebDriverUtils.download_report(driver, '/test/report.pdf')
                RemoteWebDriverUtils.download_attachment(driver, "video", '/test/video.flv')
                RemoteWebDriverUtils.download_attachment(driver, "image", '/test/image.jpg')
                
                driver.quit()
            
        print('Run ended')

if __name__ == "__main__":
    print('Run started')
    AppiumTest().main()
    
